src/lib.rs

```diff
@@ -432,7 +432,100 @@ impl Rdma {
         self.agent
             .as_ref()
             .ok_or_else(|| io::Error::new(io::ErrorKind::Other, "Agent is not ready"))?
-            .send_data(lm)
+            .send_data(lm, None)
```

Comment:
reuse `send_with_imm` function?

Reply:
`send_with_imm`  forces the user to fill in a u32, so reuse is not appropriate here.